docs/design/segments.md

```diff
@@ -23,231 +23,192 @@ title: "Segments"
-decrease query latency by scanning only those columns actually needed
-for a query.  There are three basic column types: the timestamp
-column, dimension columns, and metric columns, as illustrated in the
-image below:
+If you reindex the data with a new schema, Druid allocates a new version ID to the newly created segments:
+
+```
+foo_2015-01-01/2015-01-02_v2_0
+foo_2015-01-01/2015-01-02_v2_1
+foo_2015-01-01/2015-01-02_v2_2
+```
+
+## Segment file structure
+
+Segment files are *columnar*: the data for each column is laid out in
+separate data structures. By storing each column separately, Druid decreases query latency by scanning only those columns actually needed for a query.  There are three basic column types: timestamp, dimensions, and metrics:
 
 ![Druid column types](../assets/druid-column-types.png "Druid Column Types")
 
-The timestamp and metric columns are simple: behind the scenes each of
-these is an array of integer or floating point values compressed with
-LZ4. Once a query knows which rows it needs to select, it simply
-decompresses these, pulls out the relevant rows, and applies the
-desired aggregation operator. As with all columns, if a query doesn’t
-require a column, then that column’s data is just skipped over.
+Timestamp and metrics type columns are arrays of integer or floating point values compressed with
+[LZ4](https://github.com/lz4/lz4-java). Once a query identifies which rows to select, it decompresses them, pulls out the relevant rows, and applies the
+desired aggregation operator. If a query doesn’t require a column, Druid skips over that column's data.
 
 Dimensions columns are different because they support filter and
 group-by operations, so each dimension requires the following
 three data structures:
 
-1. A dictionary that maps values (which are always treated as strings) to integer IDs,
-2. A list of the column’s values, encoded using the dictionary in 1, and
-3. For each distinct value in the column, a bitmap that indicates which rows contain that value.
-
-
-Why these three data structures? The dictionary simply maps string
-values to integer ids so that the values in \(2\) and \(3\) can be
-represented compactly. The bitmaps in \(3\) -- also known as *inverted
-indexes* allow for quick filtering operations (specifically, bitmaps
-are convenient for quickly applying AND and OR operators). Finally,
-the list of values in \(2\) is needed for *group by* and *TopN*
-queries. In other words, queries that solely aggregate metrics based
-on filters do not need to touch the list of dimension values stored in \(2\).
+- Dictionary: Maps values (which are always treated as strings) to integer IDs, allowing compact representation of the list and bitmap values.
+- List: The column’s values, encoded using the dictionary. Required for GROUP BY and TOPN queries. These operators allow queries that solely aggregate metrics based on filters to run without accessing the list of values.
```

Comment:
I think we treat these as GroupBy and TopN https://druid.apache.org/docs/0.22.1/development/extensions-contrib/distinctcount.html